test/unit/lib/GitHub.test.js

```diff
@@ -100,25 +100,58 @@ describe('GitHub interface', () => {
 
       const githubInstance = await new GitHub(req.params)
 
+      let exception = 'thrown'
       try {
         await githubInstance.readFile(filePath)
+        exception = 'not thrown'
       } catch (err) {
         expect(err._smErrorCode).toEqual('GITHUB_READING_FILE')
       }
-
+      expect(exception).toBe('thrown')
       expect(scope.isDone()).toBe(true)
     })
 
-    test('returns an error if parsing fails for the given file', async () => {
+    test('returns an error if the config file cannot be read', async () => {
       const filePath = 'path/to/file.yml'
       const githubInstance = await new GitHub(req.params)
 
+      let exception = 'thrown'
       try {
         await githubInstance.readFile(filePath)
+        exception = 'not thrown'
+      } catch (err) {
+        expect(err._smErrorCode).toEqual('GITHUB_READING_FILE')
```

Comment:
So the old test was looking for PARSING_ERROR (but gets no error due to the error handling bug), but the test is actually trying to read a non-existent file, so it should get GITHUB_READING_FILE error instead: parsing errors are reserved for problems that occur when the file was actually read but didn't parse correctly.